### PR TITLE
Handle when unable to detect host MTU

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: dataplane/linux/int_dataplane.go [hunk 1]
```diff
@@ -801,6 +801,13 @@ func findHostMTU(matchRegex *regexp.Regexp) (int, error) {
 			smallest = l.Attrs().MTU
 		}
 	}
+
+	if smallest == 0 {
+		// We failed to find a usable interface. Default the MTU of the host
+		// to 1460 - the smallest among common cloud providers.
+		log.Warn("Failed to auto-detect host MTU, assuming 1460")
```
Comment (fasaxc) [inline dataplane/linux/int_dataplane.go:808]: You could improve this log by saying something like "Failed to auto-detect host's MTU, no interfaces matched the MTU data interface pattern; to use auto MTU detection, set <param name> to match your host's interfaces."

**Review by fasaxc — CHANGES_REQUESTED** [review]
Could do with an FV test, should just be a matter of setting the interface pattern to garbage and checking ht MTU file shows up.

**Comment by caseydavenport** [comment]
@fasaxc thanks for the review! I've added a test

File: dataplane/linux/int_dataplane.go [hunk 2]
```diff
@@ -801,6 +801,13 @@ func findHostMTU(matchRegex *regexp.Regexp) (int, error) {
 			smallest = l.Attrs().MTU
 		}
 	}
+
+	if smallest == 0 {
+		// We failed to find a usable interface. Default the MTU of the host
+		// to 1460 - the smallest among common cloud providers.
+		log.Warn("Failed to auto-detect host MTU - no interfaces matched the MTU interface pattern. To use auto-MTU, set to match your host's interfaces")
```
Comment (fasaxc) [inline dataplane/linux/int_dataplane.go:808]: :facepalm: My comment got mangled, Tried to say `...set <name of parameter> to match...`

**Review by fasaxc — CHANGES_REQUESTED** [review]

File: dataplane/linux/int_dataplane.go (hunk 2 above)
Comment (caseydavenport) [reply dataplane/linux/int_dataplane.go]: lol right, fixed it.

**Comment by caseydavenport** [comment]
@fasaxc I think this is good to go now - I'm going to merge so I can get it into a pin update and into QA, but I'll keep an eye out for any more comments.

**Pull request merged**